rtos/source/Thread.cpp

```diff
@@ -81,8 +81,10 @@ osStatus Thread::start(mbed::Callback<void()> task)
     }
 
     if (_attr.stack_mem == nullptr) {
-        _attr.stack_mem = new uint32_t[_attr.stack_size / sizeof(uint32_t)];
-        MBED_ASSERT(_attr.stack_mem != nullptr);
+        _attr.stack_mem = new (std::nothrow) uint32_t[_attr.stack_size / sizeof(uint32_t)];
+        if (_attr.stack_mem == nullptr) {
+            return osErrorNoMemory;
```

Comment:
We should `_mutex.unlock();` before returning here, otherwise we never free the lock. Do we have an autolock in Mbed? Using an autolock can help avoid these sorts of programming errors.

Reply:
Yes, `ScopedLock`

Reply:
Cool!